build.gradle

```diff
@@ -961,6 +963,53 @@ project(':streams') {
   }
 }
 
+project(':streams:streams-scala') {
+  println "Building project 'streams-scala' with Scala version ${versions.scala}"
+  apply plugin: 'scala'
+  archivesBaseName = "kafka-streams-streams-scala"
+
+  // use experimental flag to enable SAM type conversions in compiler
+  if (versions.baseScala == '2.11') {
+    tasks.withType(ScalaCompile) {
+      scalaCompileOptions.additionalParameters += ["-Xexperimental"]
+    }
+  }
+
+  dependencies {
+    compile project(':streams')
+
+    compile libs.scalaLibrary
+
+    testCompile project(':core')
+    testCompile libs.avro4s
```

Comment:
It's only to demonstrate custom Serdes. We picked up Avro since (AFAIR) @guozhangwang suggested this example in one of the earlier PR discussions. This example goes to show that custom serdes can be handled as seamlessly as primitive ones.

Reply:
I'm a bit on the fence for introducing Avro as "the one" serde in our demonstration examples rather than keeping Kafka and Avro separate, since there are many protobufs / etc fans in the community. 

How about adding avro examples in eco-system repos, e.g. in Lightbend / Confluent / etc's own examples repo They can add their own example?

Reply:
Cool .. then we can remove the test `StreamToTableJoinScalaIntegrationTestImplicitSerdesWithAvro` and the dependencies from `build.gradle` .. ok ?

Reply:
Sounds good.